python/ray/tune/function_runner.py

```diff
@@ -122,9 +124,7 @@ def create_perm_checkpoint(checkpoint_dir, logdir, step):
         return perm_checkpoint_dir
 
 
-class _StatusReporter:
-    """Object passed into your function that you can report status through."""
-
+class TuneSession(Session):
```

Comment:
I don't like that this class exposes too much through its APIs. One should only use `report`, `trial_id`, `get_resources` etc through TuneSession. One should not need to use `reset`, `make_checkpoint_dir ` and `set_checkpoint` etc.
One solution is to have TuneSession(Session) interface and _StatusReporter(TuneSession) that implements it. This way, we make clear what is guaranteed by a TuneSession interface.
The other solution is just to mark `reset`, `make_checkpoint_dir ` and `set_checkpoint` with underscore.

Reply:
Same. By having an explicit TrainSession interface for user to work with.